src/lib/network/network_autoyast.rb

```diff
@@ -242,27 +263,25 @@ def assign_udevs_to_devs(udev_rules)
         next if !valid_rename_udev_rule?(rule)
         key.downcase!
 
+        # find item which matches to the given rule definition
         item, matching_item = LanItems.Items.find do |_, i|
           i["hwinfo"]["busid"].downcase == key || i["hwinfo"]["mac"].downcase == key
         end
         next if !matching_item
 
-        # for logging only
         name_from = matching_item["ifcfg"] || matching_item["dev_name"]
         log.info("Matching device found - renaming <#{name_from}> -> <#{name_to}>")
 
-        # selecting according device name is unreliable (selects only in between configured devices)
-        LanItems.current = item
+        # find rule in collision
+        colliding_item, _item_map = LanItems.Items.find do |i, _|
+          LanItems.GetDeviceName(i) == name_to
+        end
```

Comment:
so if I get code it always find collision? does here missing check for nil? I would like to see test case also for this method.